packages/shared/lib/services/notification/slack.service.ts

```diff
@@ -485,35 +488,60 @@ export class SlackService {
      * and if so add the connection id to the connection list.
      */
     async addFailingConnection(nangoConnection: NangoConnection, name: string, type: string): Promise<ServiceResponse<NotificationResponse>> {
-        const isOpen = await this.hasOpenNotification(nangoConnection, name, type);
+        return await db.knex.transaction(async (trx) => {
+            const isOpen = await this.hasOpenNotification(nangoConnection, name, type, trx);
```

Comment:
the problem seems to be a race condition around 
```
const isOpen =  await this.hasOpenNotification(...)
if(!isOpen) {
    await this.createNotification(...)
}
```
where concurrent executions can enter the `if`. A mutex should solve the problem here. What do you think?

Reply:
Yup, updated.